x/uibc/quota/keeper/quota.go

```diff
@@ -0,0 +1,267 @@
+		var quotaOfIBCDenom uibc.Quota
+		if err := quotaOfIBCDenom.Unmarshal(iter.Value()); err != nil {
+			return nil, err
+		}
+
+		quotaOfIBCDenoms = append(quotaOfIBCDenoms, quotaOfIBCDenom)
+	}
+
+	return quotaOfIBCDenoms, nil
+}
+
+// GetQuotaByDenom retunes the rate limits of ibc denom.
+func (k Keeper) GetQuotaByDenom(ctx sdk.Context, ibcDenom string) (*uibc.Quota, error) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(uibc.KeyTotalOutflows(ibcDenom))
+	if bz == nil {
+		return nil, uibc.ErrNoQuotaForIBCDenom
+	}
+
+	var quotaOfIBCDenom uibc.Quota
+	if err := k.cdc.Unmarshal(bz, &quotaOfIBCDenom); err != nil {
+		return nil, err
+	}
+
+	return &quotaOfIBCDenom, nil
+}
+
+// SetDenomQuotas save the updated token quota.
+func (k Keeper) SetDenomQuotas(ctx sdk.Context, quotas []uibc.Quota) error {
+	for _, quotaOfIBCDenom := range quotas {
+		if err := k.SetDenomQuota(ctx, quotaOfIBCDenom); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// SetDenomQuota save the quota of denom into store.
+func (k Keeper) SetDenomQuota(ctx sdk.Context, quotaOfIBCDenom uibc.Quota) error {
+	store := ctx.KVStore(k.storeKey)
+	key := uibc.KeyTotalOutflows(quotaOfIBCDenom.IbcDenom)
+	bz, err := k.cdc.Marshal(&quotaOfIBCDenom)
+	if err != nil {
+		return err
+	}
+	store.Set(key, bz)
+
+	return nil
+}
+
+// GetTotalOutflowSum returns the total outflow of ibc-transfer amount.
+func (k Keeper) GetTotalOutflowSum(ctx sdk.Context) sdk.Dec {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(uibc.KeyPrefixTotalOutflows)
+	return sdk.MustNewDecFromStr(string(bz))
+}
+
+// SetTotalOutflowSum save the total outflow of ibc-transfer amount.
+func (k Keeper) SetTotalOutflowSum(ctx sdk.Context, amount sdk.Dec) error {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(uibc.KeyPrefixTotalOutflows, []byte(amount.String()))
+	return nil
+}
+
+// SetExpire save the quota expires time of ibc denom into store.
+func (k Keeper) SetExpire(ctx sdk.Context, expires time.Time) error {
+	store := ctx.KVStore(k.storeKey)
+	bz, err := expires.MarshalBinary()
+	if err != nil {
+		return err
+	}
+
+	store.Set(uibc.KeyPrefixQuotaExpires, bz)
+
+	return nil
+}
+
+// GetExpire returns ibc-transfer quota expires time.
+func (k Keeper) GetExpire(ctx sdk.Context) (*time.Time, error) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(uibc.KeyPrefixQuotaExpires)
+	if bz == nil {
+		return nil, nil
+	}
+	quotaExpires := ctx.BlockTime()
+	if err := quotaExpires.UnmarshalBinary(bz); err != nil {
+		return &quotaExpires, err
+	}
+	return &quotaExpires, nil
+}
+
+// ResetQuota will reset the ibc-transfer quotas
+func (k Keeper) ResetQuota(ctx sdk.Context) error {
+	qd := k.GetParams(ctx).QuotaDuration
+	newExpires := ctx.BlockTime().Add(qd)
+	if err := k.SetExpire(ctx, newExpires); err != nil {
+		return err
+	}
+
+	if err := k.SetTotalOutflowSum(ctx, sdk.NewDec(0)); err != nil {
+		return err
+	}
+
+	quotaOfIBCDenoms, err := k.GetQuotaOfIBCDenoms(ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
no need to deserialize. Instead, let's iterate over the keys and overwrite the value directly.

Reply:
we are storing serialized quota struct into store